### PR TITLE
SIP-32: Moving frontend code to the base of the repo

### LINKED ISSUE #8805: [SIP-32] Proposal for moving frontend assets to the base of the repo
### Motivation
Currently frontend assets are embedded in the superset package under `superset/assets/`, which is symlinked in`superset/assets/static`. This is really confusing for newcomers to the project as it's unclear why code appears to be duplicated. Additionally this is a UNIX symlink so the link isn't created on Windows machines. 

Moreover is seems there's code (eg, `MANIFEST.in`, `Dockerfile` ) that tries to work around the fact that the entire frontend exists in the superset module and most of these files are not needed during runtime. Things like whitelisting files and removing files after builds are workarounds to the actual issue and only add complexity to the problem. 

### Proposed Change
- Move the frontend code into a top level directory, something like `frontend` or `client`
- Configure webpack to output the built/dist files to `superset/assets/dist`. This can include the package.json (needed for version info) and any other necessary runtime files. 
- remove the symlinks. 
- update build scripts
- make sure everything works (local builds, docker builds, python package, CI, etc.)
- update documentation. 

### New or Changed Public Interfaces
[no changes]

### New dependencies
[no changes]

### Migration Plan and Compatibility

No major/breaking changes should be necessary. The bulk of the work would be around making sure no existing builds/scripts break and documentation is updated. Since most of the changes will only be moving files, rebases should be automatic, though this will probably require a lot of existing branches to be rebased. 

### Rejected Alternatives
1. Remove the symlink and move everything to `superset/static/assets` and continue to support the whitelisting in `MANIFEST.in` and the cleanup in `Dockerfile` and possibly other places. 
2. Continue to support the current file organization and continue to field questions such as: https://github.com/apache/incubator-superset/issues/8804

### PR DESCRIPTION
**Pull request by suddjian**
### CATEGORY

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Implements changes outlined in [SIP-32](https://github.com/apache/incubator-superset/issues/8805).

Removes the need for a symlink from `superset/static` to `superset/assets` by moving "assets" to a new top-level `/superset-frontend` directory.

Webpack now outputs into the `superset/static/assets` folder, copying any necessary assets.

Simplifies `MANIFEST.in`. Excluding frontend source is no longer necessary since the source doesn't live in the backend directory.

Removing two more symlinks between frontend assets and docs, instead copying built files to the `static` folder. This was necessary because copying the built files breaks these symlinks, but it also cleans up the directory structure.

### TEST PLAN

Running `npm run build`, `npm run dev`, and `npm run dev-server` all seem to work correctly. 
Building the tarball seems to work fine, but could use additional verification as I'm not sure what all the failure states there might be.
Docker build seems to work fine, but could use more verification here as well.
Whether CI still works remains to be seen, this PR will test that.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Closes #8805 
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@nytai @willbarrett @mistercrunch @craig-rueda @rusackas @dpgaspar @DiggidyDave @villebro

### CODE REVIEW

File: superset-frontend/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "cover": "jest --coverage",
     "dev": "webpack --mode=development --colors --progress --debug --watch",
     "dev-server": "node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=development --progress",
-    "prod": "node --max_old_space_size=4096 webpack --mode=production --colors --progress",
+    "prod": "node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=production --colors --progress",
```
Comment (nytai) [inline superset-frontend/package.json]: do we want to be running the webpack-dev-server or just webpack for the prod build?

File: superset/static/assets [hunk 2]
```diff
@@ -1 +0,0 @@
-../assets
```
Comment (nytai) [inline superset/static/assets]: Finally!!!!!!!!!!! 😄 😄 😄

File: superset-frontend/package.json (hunk 1 above)
Comment (suddjian) [reply superset-frontend/package.json]: oops good catch. Copy/paste oversight there.

**Review by mistercrunch — APPROVED** [review]

**Comment by mistercrunch** [comment]
giphy

**Pull request merged**